cli/tests/activate.bats

```diff
@@ -4604,3 +4604,101 @@ EOF
   assert_output --partial "zsh: the outer alias is defined"
   assert_output --partial "zsh: the current alias is inner"
 }
+
+in_place_doesnt_override_user_initiated() {
+  shell="${1?}"
+  shell_path="$(which "$shell")"
+  export _FLOX_SHELL_FORCE="$shell_path"
+
+  PROJECT_DIR_CANONICAL="$(realpath "$PROJECT_DIR")"
+  PROJECT_DEFAULT="default-rc"
+  PROJECT_USER="user-initiated"
+  MODE="dev"
+
+  run "$FLOX_BIN" init -d "$PROJECT_DEFAULT"
+  run "$FLOX_BIN" init -d "$PROJECT_USER"
+
+  IN_PLACE_COMMAND="${FLOX_BIN} activate -m ${MODE} -d ${PROJECT_DEFAULT}"
+  # Delay evaluation until within activation. Must be shell-agnostic.
+  # shellcheck disable=SC2016
+  ECHO_COMMANDS='echo FLOX_ENV: $FLOX_ENV
+echo FLOX_ENV_CACHE: $FLOX_ENV_CACHE
+echo FLOX_ENV_PROJECT: $FLOX_ENV_PROJECT
+echo FLOX_ENV_DESCRIPTION: $FLOX_ENV_DESCRIPTION
+echo ---'
+
+  case "$shell" in
+    bash)
+      cat > "${HOME}/.bashrc" <<EOF
+eval "\$(${IN_PLACE_COMMAND})"
+${ECHO_COMMANDS}
+EOF
+      ;;
+    fish)
+      cat > "${HOME}/.config/fish/config.fish" <<EOF
+${IN_PLACE_COMMAND} | source
+${ECHO_COMMANDS}
+EOF
+      ;;
+    tcsh)
+      cat > "${HOME}/.tcshrc" <<EOF
+eval "\`${IN_PLACE_COMMAND}\`"
+${ECHO_COMMANDS}
+EOF
+      ;;
+    zsh)
+      cat > "${HOME}/.zshenv" <<EOF
+eval "\$(${IN_PLACE_COMMAND})"
+${ECHO_COMMANDS}
+EOF
+      ;;
+    *)
+      echo "Unsupported shell: ${shell}"
+      exit 1
+      ;;
+  esac
+
+  if [ "$shell" = "tcsh" ]; then
+    # Our command quoting is broken for tcsh.
+    # shellcheck disable=SC2086
+    run "$FLOX_BIN" activate -m "$MODE" -d "${PROJECT_USER}" -- "${shell_path}" -c ${ECHO_COMMANDS}
+  else
+    run "$FLOX_BIN" activate -m "$MODE" -d "${PROJECT_USER}" -- "${shell_path}" -c "${ECHO_COMMANDS}"
+  fi
+  assert_success
+
+  # Assert on both activations to ensure that we really got an in-place activation first.
+  # TODO: Remove --partial when duplicate activations are fixed for zsh and fish.
```

Comment:
Will create a follow-up ticket for this if no one has any quick ideas.

Reply:
Taking a quick look at this, but no need to block

Reply:
Bitten again, not for the first or the last time. I've converted them to unquoted args. (will squash)